File: 백준/Silver/1735. 분수 합/분수 합.py
import math

def find_gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def simplify_fraction(numerator, denominator):
    gcd = find_gcd(numerator, denominator)
    simplified_numerator = numerator // gcd
    simplified_denominator = denominator // gcd
    return simplified_numerator, simplified_denominator

a, b = map(int, input().split())
c, d = map(int, input().split())

A, B = a * d + c * b, d * b
simple_A, simple_B = simplify_fraction(A, B)
print(simple_A, simple_B, sep=" ")

File: 백준/Silver/1735. 분수 합/분수 합.lua
function gcd(a, b)
    while b ~= 0 do
        a, b = b, a % b
    end
    return a
end

function simplify_fraction(A, B)
    C = gcd(A, B)
    simple_a = A // C
    simple_b = B // C
    return simple_a, simple_b
end

local a, b = io.read("*n", "*n")
local c, d = io.read("*n", "*n")

A, B = a * d + c * b, d * b
x, y = simplify_fraction(A, B)
print(x .. ' ' .. y)